### PR TITLE
Amazon scraping service implemented to get the details of a product b…

### PR DESCRIPTION
**Pull request by christian-barnabe-chabi**
### What this PR does (required):
- Amazon scraping service implemented to get the details of a product by passing its url

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

```js
const { getProductDetail } = require('../services/amazonScrapingService');
getProductDetail(
  "https://www.amazon.com/TCL-32-720p-ROKU-Smart/dp/B088S3V3R4/ref=sr_1_1?dchild=1&field-shipping_option-bin=3242350011&pf_rd_i=16225009011&pf_rd_m=ATVPDKIKX0DER&pf_rd_p=85a9188d-dbd5-424e-9512-339a1227d37c&pf_rd_r=GCPE3FVDXAAZ1G0434VD&pf_rd_s=merchandised-search-5&pf_rd_t=101&qid=1622646074&rnid=1266092011&s=electronics&sr=1-1"
).then((data) => console.log(data));
```

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/services/amazonScrapingService.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        );
+
+        const features = featuresLiElem.map((liElem) => {
+          return liElem.innerText;
+        });
+
+        let price = "unavailable";
+
+        try {
+          price = document.querySelector("span#priceblock_ourprice").innerText;
```
Comment (christian-barnabe-chabi) [inline server/services/amazonScrapingService.js:25]: I'll check if this works for regular and sale price.

Comment (christian-barnabe-chabi) [reply server/services/amazonScrapingService.js:25]: It seems that the span#priceblock_ourprice selector is for both regular and sale prices. In fact, this selector is for the current sale price

**Review by sina-jamshidi — APPROVED** [review]

**Comment by marouane-erouhi** [comment]
some items use `priceblock_saleprice` as the tag name ([example](https://www.amazon.com/Led-Strip-Lights-Keepsmile-Decoration/dp/B093CGRVM6/ref=sxin_9_ac_d_mf_rm?ac_md=0-0-bGVkIHN0cmlw-ac_d_rm&cv_ct_cx=led+strip&dchild=1&keywords=led+strip&pd_rd_i=B093CGRVM6&pd_rd_r=435958d0-caa7-4e70-8b00-0cce0cf22771&pd_rd_w=IwKmY&pd_rd_wg=eZhpx&pf_rd_p=d00be302-86a9-4ea9-9240-f5ec923e66bc&pf_rd_r=RCRKXA9F9NH7JETF0VE6&psc=1&qid=1622827859&sr=1-1-849f3c3a-785c-4812-aab6-3d7cb06022f2))instead of `priceblock_ourprice`

`"span[id*='priceblock']"` seems to work for both

**Comment by christian-barnabe-chabi** [comment]
> - some items use `priceblock_saleprice` as the tag name ([example](https://www.amazon.com/Led-Strip-Lights-Keepsmile-Decoration/dp/B093CGRVM6/ref=sxin_9_ac_d_mf_rm?ac_md=0-0-bGVkIHN0cmlw-ac_d_rm&cv_ct_cx=led+strip&dchild=1&keywords=led+strip&pd_rd_i=B093CGRVM6&pd_rd_r=435958d0-caa7-4e70-8b00-0cce0cf22771&pd_rd_w=IwKmY&pd_rd_wg=eZhpx&pf_rd_p=d00be302-86a9-4ea9-9240-f5ec923e66bc&pf_rd_r=RCRKXA9F9NH7JETF0VE6&psc=1&qid=1622827859&sr=1-1-849f3c3a-785c-4812-aab6-3d7cb06022f2))instead of `priceblock_ourprice`
> 
> - `"span[id*='priceblock']"` seems to work for both
> 
> - also `puppeteer` is not in package.json

Oh, cool. Ty. I'll check it

**Comment by christian-barnabe-chabi** [comment]
> * priceblock_saleprice

I tried using `span[id*='priceblock']` but it selected a lot of blocks

**Pull request merged**